libsolidity/ast/AST.cpp

```diff
@@ -84,13 +84,41 @@ SourceUnitAnnotation& SourceUnit::annotation() const
 	return dynamic_cast<SourceUnitAnnotation&>(*m_annotation);
 }
 
-string Declaration::sourceUnitName() const
+std::set<SourceUnit const*> SourceUnit::referencedSourceUnits(bool _flatten, std::set<SourceUnit const*> _skipList) const
+{
+	std::set<SourceUnit const*> sourceUnits;
+	for (auto const& node: nodes())
+	{
+		if (auto const* importDirective = dynamic_cast<ImportDirective const*>(node.get()))
+		{
+			auto annotation = importDirective->annotation();
```

Comment:
We actually only use the `sourceUnit` from the annotation, so we could store this instead in a local variable.